Update README (GA) (#22)
slack-samples/deno-github-functions: README.md

> # Workflows for GitHub Sample App

Reviewer: A very welcomed change to this header 🙌

> This sample demonstrates how to create automation in Slack using GitHub.

Reviewer: ```suggestion
This sample demonstrates automations in Slack that interact with the GitHub API.
```

Feel free to adjust this or ignore, but I think guides do more wholistic job of demonstrating "how to create" than the sample. Maybe we also include [the guide](https://api.slack.com/tutorials/tracks/create-github-issues-in-workflows) in the README?

**Guide Outline**:

- [Included Workflows](#included-workflows)
- [Setup](#setup)
  - [Install the Slack CLI](#install-the-slack-cli)
  - [Clone the Sample App](#clone-the-sample-app)
  - [Register a GitHub App](#register-a-github-app)
  - [Configure Outgoing Domains](#configure-outgoing-domains)
- [Creating Triggers](#creating-triggers)

> - [Datastores](#datastores)

Reviewer: ```suggestion
```

Some comments on this are below!

trigger that generates a **Shortcut URL** which, when posted in a channel or
added as a bookmark, becomes a link. When clicked, the link trigger will run the
associated workflow.

Link triggers are _unique to each installed version of your app_. This means
that Shortcut URLs will be different across each workspace, as well as between
[locally run](#running-your-project-locally) and
[deployed apps](#deploying-your-app).

With link triggers, after selecting a workspace and environment, the output
provided will include a Shortcut URL. Copy and paste this URL into a channel as
a message, or add it as a bookmark in a channel of the workspace you selected.
Interacting with this link will run the associated workflow.

**Note: triggers won't run the workflow unless the app is either running locally
or deployed!**

### Manual Trigger Creation

To manually create a trigger, use the following command:

```zsh
$ slack trigger create --trigger-def triggers/create_new_issue_shortcut.ts
```

## Datastores

For storing data related to your app, datastores offer secure storage on Slack
infrastructure. The use of a datastore requires the

> `datastore:write`/`datastore:read` scopes to be present in your manifest.

Reviewer: I believe this section can be removed for this sample! 🚫🗄️!

## Testing

For an example of how to test a function, see `functions/create_issue_test.ts`.
Test filenames should be suffixed with `_test`.

Run all tests with `deno test`:

```zsh
$ deno test
```

## Deploying Your App

Once development is complete, deploy the app to Slack infrastructure using
`slack deploy`:

```zsh
$ slack deploy
```

When deploying for the first time, you'll be prompted to
[create a new link trigger](#creating-triggers) for the deployed version of your
app. When that trigger is invoked, the workflow should run just as it did when
developing locally (but without requiring your server to be running).

## Viewing Activity Logs

Activity logs of your application can be viewed live and as they occur with the
following command:

```zsh
$ slack activity --tail
```

> ## Project Structure

Reviewer: Could we also include the `external_auth/` section found on [the timesheet approval sample](https://github.com/slack-samples/deno-timesheet-approval#external_auth)?

### `.slack/`

Contains `apps.dev.json` and `apps.json`, which include installation details for
development and deployed apps.

### `datastores/`

[Datastores](https://api.slack.com/automation/datastores) securely store data
for your application on Slack infrastructure. Required scopes to use datastores

> include `datastore:write` and `datastore:read`.

Reviewer: This sample doesn't include this, so I think this section can be removed!
